ca/certificate-authority.go

```diff
@@ -243,38 +240,13 @@ func NewCertificateAuthorityImpl(
 	return ca, nil
 }
 
-// checkHSMFault checks whether there has been an HSM fault observed within the
-// timeout window.  CA methods that use the HSM should call this method right
-// away, to minimize the performance impact of HSM outages.
-func (ca *CertificateAuthorityImpl) checkHSMFault() error {
-	ca.hsmFaultLock.Lock()
-	defer ca.hsmFaultLock.Unlock()
-
-	// If no timeout is set, never gate on a fault
-	if ca.hsmFaultTimeout == 0 {
-		return nil
-	}
-
-	now := ca.clk.Now()
-	timeout := ca.hsmFaultLastObserved.Add(ca.hsmFaultTimeout)
-	if now.Before(timeout) {
-		err := core.ServiceUnavailableError("HSM is unavailable")
-		ca.log.WarningErr(err)
-		ca.stats.Inc(metricHSMFaultRejected, 1, 1.0)
-		return err
-	}
-	return nil
-}
-
 // noteHSMFault updates the CA's state with regard to HSM faults.  CA methods
 // that use an HSM should pass errors that might be HSM errors to this method.
 func (ca *CertificateAuthorityImpl) noteHSMFault(err error) {
-	ca.hsmFaultLock.Lock()
-	defer ca.hsmFaultLock.Unlock()
-
 	if err != nil {
-		ca.stats.Inc(metricHSMFaultObserved, 1, 1.0)
-		ca.hsmFaultLastObserved = ca.clk.Now()
+		if cfErr, ok := err.(*cferr.Error); ok && int(cfErr.ErrorCode/1000) == 1 {
```

Comment:
This deserves a comment. Why is this range of errors specific to HSM problems? Also why do you need to cast to int?